### DISCUSSION TITLE
Installation problem with pycharm

### DISCUSSION THREAD

**Discussion by oscarhx** [PMEAL/OpenPNM#2321] [Q&A]
Hello,
If using pycharm to install openpnm from the sourcecode, can you elaborate the following command: 
pip install -e 'path/to/downloaded/files', for example, the source code is in file path F:/openpnm-dev
since  using pip to install pypardiso fails on Python 3.7+.

**Comment by jgostick** [comment]
Hi @oscarhx, installation from source is easier if you just navigate to the directory where the code is (``c:\ cd F:\openpnm-dev``), then run ``pip install -e .``  The ``.`` indicates 'the current directory'.

Once you've install it, you may have to fix the pardiso problem, which you can do by first removing ``mkl`` and ``intel-openmp`` with conda ``conda remove --force mkl intel-openmp``.  The ``force`` option is crucial so that you don't uninstall all the other deps, just those two. Then you'll want to reinstall version of those packages that actually work on windows using ``conda install mkl=2021.4.0 intel-openmp=2021.4.0``.  Lastly, since you installed openpnm with pip rather than conda you'll need to get the pypardiso package with ``conda install -c conda-forge pypardiso``.  The ``-c conda-force`` argument tells conda to look on conda-forge rather than the normal conda repo.
